### PR TITLE
Prepare release for serialization modules

### PR DESCRIPTION
**Pull request by alzimmermsft**

### CODE REVIEW

**Comment by alzimmermsft** [comment]
Cherry picked a few commits, that won't be included in this release, that fix flaky tests so CI doesn't randomly fail.

**Comment by alzimmermsft** [comment]
> 
> 
> Which packages are going to release?
> All serializer impl packages are missing `uses` in module-info.java

Planning on releasing `azure-core-serializer-json-jackson`, `azure-core-serializer-json-gson`, `azure-core-serializer-avro-apache`, and `azure-core-serializer-avro-jackson`.

Good catch on the missing `uses`, I'll update it.

**Comment by alzimmermsft** [comment]
@sima-zhu went read up on service providers, only the module using the ServiceLoader needs to have `uses` which `azure-core-experimental` does. It looks like the HttpClient modules don't need to have the `uses` in their `module-info`.

cc: @srnagar

File: sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/polling/LROPollerTests.java [hunk 1]
```diff
@@ -474,7 +474,7 @@ public String getName() {
             } catch (InterruptedException e) {
                 //
             }
-            Assertions.assertEquals(count, getCallCount.get());
+            //Assertions.assertEquals(count, getCallCount.get());
```
Comment (srnagar) [inline sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/polling/LROPollerTests.java:477]: Any reason for commenting this out? Looks like the poller tests should not be impacted by this PR.

File: sdk/storage/azure-storage-blob-changefeed/src/test/java/com/azure/storage/blob/changefeed/ChangefeedNetworkTest.groovy [hunk 2]
```diff
@@ -73,6 +74,7 @@ class ChangefeedNetworkTest extends APISpec {
 
     @Unroll
     @Requires( { playbackMode() })
+    @Ignore("flaky in CI")
```
Comment (srnagar) [inline sdk/storage/azure-storage-blob-changefeed/src/test/java/com/azure/storage/blob/changefeed/ChangefeedNetworkTest.groovy:77]: Is there a GitHub issue to fix the flakiness?

File: sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/polling/LROPollerTests.java (hunk 1 above)
Comment (alzimmermsft) [reply sdk/core/azure-core-management/src/test/java/com/azure/core/management/implementation/polling/LROPollerTests.java:477]: These have been fixed and merged in after the commit this branch was checked out from. Cherry picked to make it so CI fails less.

File: sdk/storage/azure-storage-blob-changefeed/src/test/java/com/azure/storage/blob/changefeed/ChangefeedNetworkTest.groovy (hunk 2 above)
Comment (alzimmermsft) [reply sdk/storage/azure-storage-blob-changefeed/src/test/java/com/azure/storage/blob/changefeed/ChangefeedNetworkTest.groovy:77]: These have been fixed and merged in after the commit this branch was checked out from. Cherry picked to make it so CI fails less.

**Pull request merged**